dvc/output/base.py

```diff
@@ -389,24 +404,30 @@ def checkout(
                 )
             return None
 
+        obj = self.get_obj(filter_info=filter_info)
+        if not obj and (filter_info and filter_info != self.path_info):
+            # backward compatibility
+            return None
+
+        added = not self.exists
```

Comment:
this is what checkout() was doing, but now in an explicit way, since we care about status with output-level granularity, not finer